src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php

```diff
@@ -193,6 +195,13 @@ public function process(ContainerBuilder $container)
             }
         }
 
+        if ($container->hasDefinition('console.command.cache_pool_get')) {
+            $container->getDefinition('console.command.cache_pool_get')->replaceArgument(0, $allPoolsReference);
```

Comment:
This means that the commands will instantiate all pools, isn't it?
Maybe that's not an issue, but the rest is usually quite lazy so we might want to be lazy too here?

Reply:
The aim is to get the right cache pool from the arg given by the dev

I do not know what/how is the best way to achieve that service ref inside the command
That is why for now i pass all pools services to those commands

Also this is why i added the « $allPoolsReference » above

Reply:
Sure, that's what the code tells about this, and why I suggested to improve this :)

Reply:
@nicolas-grekas yes indeed, i may need an example of how to reference lazily those services and get them in the cli
is there any i can take a look on?

the PR in this state is more like a POC :)

Reply:
Pools are already injected lazily in the cache clearer just above (see where $allPool goes).
You can inject things in a similar way I suppose.

Reply:
i'll try this ASAP then